fdbserver/VersionedBTree.actor.cpp

```diff
@@ -5800,9 +5800,11 @@ class KeyValueStoreRedwoodUnversioned : public IKeyValueStore {
 				// Read page contents without using waits
 				bool isRoot = cur.inRoot();
 				BTreePage::BinaryTree::Cursor leafCursor = cur.popPath();
+				// we can bypass the bounds check for each key in the leaf if the entire leaf is in range
+				bool boundsCheck = leafCursor.upperBound() > keys.end;
```

Comment:
This can be >= because both query end and page upper bound are exclusive of the query results and page contents, respectively.